src/Raven.Server/Documents/Indexes/Persistence/Lucene/IndexReadOperation.cs

```diff
@@ -236,7 +274,7 @@ QueryResult CreateQueryResult(Document d)
                     if (search.TotalHits == search.ScoreDocs.Length)
                         break;
 
-                    if (returnedResults >= pageSize)
+                    if (returnedResults >= pageSize || scannedDocuments.Value == query.ScanLimit)
```

Comment:
@ayende wouldn't be safer to use `>=` instead of `==`?

Reply:
Yes

Reply:
Changed